garage/envs/mujoco/randomization/variation.py

```diff
@@ -170,17 +174,26 @@ def get_randomized_xml_model(self):
             # Check if the sampled value has the same shape with default value
             if np.array(c).shape != np.array(self._default_cache[v]).shape:
                 raise ValueError(
-                    "Sampled value you input %s don't match with default value %s in the xml node %s"
-                    % (c, self._default_cache[v], v.xpath))
+                    "Sampled value you input %s does not match with default "
+                    "value %s in the xml node %s" % (c, self._default_cache[v],
+                                                     v.xpath))
 
             if v.method == Method.COEFFICIENT:
-                e.attrib[v.attrib] = str(c * self._default_cache[v])
+                # store attrib. convert numpy print to xml-friendly print
+                e.attrib[v.attrib] = \
+                    str(c * self._default_cache[v]).strip("[],")
             elif v.method == Method.ABSOLUTE:
-                e.attrib[v.attrib] = str(c)
+                # store attrib. convert numpy print to xml-friendly print
+                e.attrib[v.attrib] = str(c).strip("[],")
             else:
                 raise ValueError("Unknown method")
 
-        return etree.tostring(self._parsed_model.getroot()).decode("ascii")
+        et = etree.ElementTree(self._parsed_model.getroot())
+        filename = os.path.join(MODEL_DIR, "varied_params.xml")
+        et.write(filename, pretty_print=True)
```

Comment:
unfortunately this makes your class no longer serializable, because your modified XML lives on disk. in order for this to be serializable, i should be able to ship this .pkl file to the moon, along with a copy of the repo which created the .pkl, and it should still work.

why do you have to write a file anyway? is there a reason the in-memory version can't be serialized?